dj/service_clients.py

```diff
@@ -82,3 +84,46 @@ def get_columns_for_table(
             Column(name=column["name"], type=ColumnType(column["type"]))
             for column in table_columns
         ]
+
+    def submit_query(  # pylint: disable=too-many-arguments
+        self,
+        engine: str,
+        engine_version: str,
+        submitted_query: str,
+        catalog: Optional[str] = "default",
+        async_: Optional[bool] = False,
+    ) -> QueryWithResults:
+        """
+        Submit a query to the query service
+        """
+        response = self.requests_session.post(
+            "/queries/",
+            json={
+                "engine": engine,
+                "engine_version": engine_version,
+                "submitted_query": submitted_query,
+                "catalog_name": catalog,
+                "async_": async_,
+            },
+        )
+        if response.status_code >= 400:
+            raise DJException(
+                message=f"Error response from query service: {response.text}",
+            )
+        query_info = response.json()
+        return QueryWithResults(**query_info)
```

Comment:
Let's make it `QueryResults` please? :) 

`QueryWithResults` feels a bit weird. It makes me want to ask for `QueryWithoutResults` as well.

Reply:
Haha, every-time I see that I think of how much you're not a fan of that class name. 😂 I have a big refactor coming up to fully complete the move to the API and wiring up DJQS. Once that's in I'll remember to come back and update the class name here and in DJQS. 💯

Reply:
@agorajek but this is a query object with results! 🙂 

It has the query information (status, progress, submitted/executed query) and the results. It's almost identical to the `Query` payload, with the addition of the result, hence `QueryWithResults`.

https://github.com/DataJunction/dj/blob/110168faedd7d052f86107ff71c0c60dc648d93c/dj/models/query.py#L114

Reply:
And for context, the idea with `QueryWithResults` is that it simplifies the query endpoint. You can a single endpoint where you can create queries (doing a `POST`), check for their status (`GET`) and fetch results (also `GET`).

Reply:
![hold-on-hmm](https://user-images.githubusercontent.com/371403/224111915-440972d2-126a-4879-b6fb-e8de9c8ef747.gif)

OK, I will not bring this subject again. If I do please add it to my performance review.